components/api/core/utils.bal

```diff
@@ -39,3 +40,120 @@ function updatePayloadWithUserInfo (json payload, string field) returns error? {
         log:printDebug(io:sprintf("Response modification failed : User information not found for user : \'%s\'", userId));
     }
 }
+
+function getUserToken(string authorizationHeader, string cookieHeader) returns string | error? {
+    string[] splittedToken = authorizationHeader.split(" ");
+    if splittedToken.length() != 2 || !(splittedToken[0].equalsIgnoreCase(constants:BEARER_HEADER)) {
+        error er = error("Did not receive the token in proper format");
+        return er;
+    }
+    string lastTokenElement = splittedToken[1];
+    string | error firstTokenElement = filter:getCookie(cookieHeader);
+    if (firstTokenElement is error) {
+        error er = error(io:sprintf("Cookie value could not be resolved. %s", firstTokenElement.reason()));
+        return er;
+    } else {
+        string token = io:sprintf("%s%s", firstTokenElement, lastTokenElement);
+        if "" == token {
+            error er = error("Did not receive any token");
+            return er;
+        }
+        return token;
+    }
+}
+
+function deleteArtifactFromRegistry(http:Request deleteArtifactReq, string orgName, string imageName, string artifactVersion) returns error? {
+    string userId = deleteArtifactReq.getHeader(constants:AUTHENTICATED_USER);
+    var token = getUserToken(deleteArtifactReq.getHeader(constants:AUTHORIZATION_HEADER),
```

Comment:
This can either be a token or an error. Hence shall we use a meaningful name ?

Reply:
We should use always meaningful names. The reason is if you get an error as the result, you are just returning the error. By looks the return statement doesn't look like an error. Please fix this all places.